### PR TITLE
Add randomised option values

### PR DESCRIPTION
**Pull request by barnabwhy**
Random option values: { type = "MULTI", choices = [Any], weights = [Number] }
Embed and file support added to message actions
Added more placeholder functionality

### CODE REVIEW

File: src/bot.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ module.exports = class Bot extends EventEmitter {
                             channel: msg.channel,
                             member: msg.member,
                             command: null,
-                            limitLog: (await this.config.get('on_message')).limit_log,
+                            event_config: await this.config.get('on_message'),
```
Comment (nanaian) [inline src/bot.js:68]: Why the `snake_case`?

File: src/bot.js [hunk 2]
```diff
@@ -356,7 +360,24 @@ module.exports = class Bot extends EventEmitter {
                     }
                 }
             }
-            action = await placeholdersInOpts(action, source)
+            for (const [key, value] of Object.entries(action)) {
+                action[key] = multiOption(value)
+            }
+            let event_config = JSON.parse(JSON.stringify(source.event_config))
```
Comment (nanaian) [inline src/bot.js:366]: Definitely worth adding a comment with the reason why you do this.

File: src/bot.js [hunk 3]
```diff
@@ -462,6 +483,7 @@ module.exports = class Bot extends EventEmitter {
                     channel: reaction.message.channel,
                     member: reaction.message.member,
                     command: null,
+                    event_config: await await pinConfig,
```
Comment (nanaian) [inline src/bot.js:486]: Double await.

File: src/bot.js [hunk 4]
```diff
@@ -484,6 +506,10 @@ module.exports = class Bot extends EventEmitter {
                 return map.hasOwnProperty(key)
             },
 
+            getRaw: key => {
+                return map[key]
+            },
```
Comment (nanaian) [inline src/bot.js:511]: Could put this on a single line and omit the braces and `return`.

File: src/bot.js (hunk 3 above)
Comment (barnabwhy) [reply src/bot.js:486]: oops

**Pull request merged**